src/dotnet/commands/dotnet-compile-native/ILCompilerInvoker.cs

```diff
@@ -43,32 +37,29 @@ private void InitializeArgs(NativeCompileSettings config)
             var coreLibsPath = Path.Combine(config.IlcSdkPath, "sdk");
             foreach (var reference in Directory.EnumerateFiles(coreLibsPath, "*.dll"))
             {
-                argsList.Add($"-r");
-                argsList.Add($"{reference}");
+                argsList.Add($"-r:{reference}");
             }
             
             // AppDep References
             foreach (var reference in config.ReferencePaths)
             {
-                argsList.Add($"-r");
-                argsList.Add($"{reference}");
+                argsList.Add($"-r:{reference}");
             }
             
             // Set Output DetermineOutFile
             var outFile = DetermineOutputFile(config);
-            argsList.Add($"-out");
-            argsList.Add($"{outFile}");
+            argsList.Add($"-o:{outFile}");
             
             // Add Mode Flag TODO
             if (config.NativeMode == NativeIntermediateMode.cpp)
             {
-                argsList.Add("-cpp");
+                argsList.Add("--cpp");
```

Comment:
We should start using `System.CommandLine` syntax rules for ILC. I know this means we have to update all docs...